src/Appwrite/Utopia/Database/Validator/Queries/Base.php

```diff
@@ -48,33 +50,43 @@ public function __construct(string $collection, array $allowedAttributes)
                 continue;
             }
 
-            $attributes[] = new Document([
+            $attributeDocument = new Document([
                 'key' => $key,
                 'type' => $attribute['type'],
                 'array' => $attribute['array'],
             ]);
+
+            $attributes[] = $attributeDocument;
+            $allAttributes[] = $attributeDocument;
         }
 
-        $attributes[] = new Document([
-            'key' => '$id',
-            'type' => Database::VAR_STRING,
-            'array' => false,
-        ]);
-        $attributes[] = new Document([
-            'key' => '$createdAt',
-            'type' => Database::VAR_DATETIME,
-            'array' => false,
-        ]);
-        $attributes[] = new Document([
-            'key' => '$updatedAt',
-            'type' => Database::VAR_DATETIME,
-            'array' => false,
-        ]);
-        $attributes[] = new Document([
-            'key' => '$sequence',
-            'type' => Database::VAR_INTEGER,
-            'array' => false,
-        ]);
+        $internalAttributes = [
+            new Document([
+                'key' => '$id',
+                'type' => Database::VAR_STRING,
+                'array' => false,
+            ]),
+            new Document([
+                'key' => '$createdAt',
+                'type' => Database::VAR_DATETIME,
+                'array' => false,
+            ]),
+            new Document([
+                'key' => '$updatedAt',
+                'type' => Database::VAR_DATETIME,
+                'array' => false,
+            ]),
+            new Document([
+                'key' => '$sequence',
+                'type' => Database::VAR_INTEGER,
+                'array' => false,
+            ])
+        ];
+
+        foreach ($internalAttributes as $attribute) {
+            $attributes[] = $attribute;
+            $allAttributes[] = $attribute;
```

Comment:
What's the difference between $attributes and $allAttributes?

Reply:
attributes only got ones allowed for filter queries (order, equals, ...)

I needed separate list that got it all.
Could be rewritten to first do allAttributes and then filter, but didnt see much benefit in it

Reply:
Where do we add to `$allAttributes` but not `$attributes`? Both loops I see add to both